docs/how-tos/typescript_consume.md

```diff
@@ -0,0 +1,161 @@
+- `relay-runtime` for core Relay functionality
+- `babel-plugin-relay` for compiling Relay queries
+
+## Setup Relay Environment
+
+We initially must set up a Relay environment which allows Relay to access the GraphQL endpoint.
+In the root of `src` we create a `RelayEnvironment.ts`. In this file we are setup up a Relay environment for making GraphQL requests.
+The `fetchFn` function abstracts the networking stack, sending a `POST` request to the GraphQL endpoint with the query and variables while also allowing for extensions like rate limiting or custom error handling as needed.
+The `createRelayEnvironment` function initializes the Relay environment by combining the network layer (responsible for fetching data) and the store (which caches the data).
+
+!!! example "RelayEnvironment.ts"
+
+    ```typescript
+    import {
+        Environment,
+        Network,
+        RecordSource,
+        Store,
+        FetchFunction,
+    } from "relay-runtime";
+
+    const HTTP_ENDPOINT = "http://127.0.0.1:8000/graphql"; // assuming you are running any of the services from how to guides
+
+    const fetchFn: FetchFunction = async (request, variables) => {
+        const response = await fetch(HTTP_ENDPOINT, {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify({
+                query: request.text,
+                variables,
+            }),
+        });
+
+        return await response.json();
+    };
+
+    function createRelayEnvironment() {
+        return new Environment({
+            network: Network.create(fetchFn),
+            store: new Store(new RecordSource()),
+        });
+    }
+
+    export const RelayEnvironment = createRelayEnvironment();
+
+    ```
+
+!!! info
+
+    Refer to the [Relay Environment Docs](https://relay.dev/docs/v1.6.2/relay-environment/) to learn more.
+
+## Generate TypeScript Types
+
+Relay generates TypeScript types automatically for your queries, based on the GraphQL schema.
+For example, if you are querying a Person type form the GraphQL schema, Relay will generate types for you.
+
+```graphql
+query PersonQuery {
+  person {
+    id
+    firstName
+    lastName
+  }
+}
+```
+
+Relay will automatically generate a TypeScript type for this Query when you run the Relay compiler.
+
+```typescript
+// __generated__/PersonQuery.graphql.ts
+
+export interface PersonQuery {
+  person: {
+    id: string;
+    firstName: string;
+    lastName: string;
+  } | null;
+}
+```
+
+## Fetch Data Using Relay
+
+In Relay, we define our GraphQL queries within components using the `useLazyLoadQuery` or `useFragment` hooks. Relay then fetches and manages data for you.
+
+To fetch data, we define a query in our component and use Relay's `useLazyLoadQuery` hook to execute it. The query is defined using the GraphQL tag provided by `babel-plugin-relay`.
+
+!!! example "Fetch Data with Relay"
+
+    ```ts
```

Comment:
Does `ts` work? Or does it need to be `typescript`?

Reply:
`ts` works